proxy.h

```diff
@@ -759,6 +759,11 @@ class proxy : public details::facade_traits<F>::direct_accessor {
   auto&& operator*() const&& noexcept requires(_Traits::has_indirection)
       { return std::forward<const typename _Traits::indirect_accessor>(ia_); }
 
+  friend void swap(proxy& lhs, proxy& rhs) noexcept(noexcept(lhs.swap(rhs)))
+      { lhs.swap(rhs); }
+  friend bool operator==(const proxy& lhs, std::nullptr_t) noexcept
```

Comment:
do we need have an overload for `nullptr == a_proxy`?

Reply:
No. Like `operator!=`, it can be synthesized as well. This change aligns with `std::move_only_function`.